docs/build_from_source.md

```diff
@@ -0,0 +1,100 @@
+# Build BladeDISC from Source in a Docker container
+
+This document introduced how to build BladeDISC from source,
+to easy the software installation and configuration, we provide a
+[Dockerfile](/docker/dev/Dockerfile) that contains requirements software
+to build and test.
+
+## Prerequisite
+
+1. Git for checking out the source code.
+1. [Nvidia Docker](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html)
+to launch Docker container on Nvidia GPU host.
+
+## Checkout the Source
+
+``` bash
+git clone git@github.com:alibaba/BladeDISC.git
+cd BladeDISC && git submodule update --init --recursive
+```
+
+## Building BladeDISC for TensorFlow Wrapper
+
+1. step0: launch a development Docker container that runs a
+development Docker image.
+
+    ``` bash
+    nvidia-docker run --rm -it -v $PWD:/disc bladedisc/bladedisc:latest-devel-cuda11.0 bash
+    ```
+
+    please goto [this website](https://hub.docker.com/r/bladedisc/bladedisc/tags?page=1&name=devel) to
+    find more images with various CUDA versions.
+
+1. step1: configuration stage.
+
+    ``` bash
+    python scripts/python/tao_build.py /opt/venv_disc -s configure --bridge-gcc default --compiler-gcc default
+    ```
+
+    - `/opt/venv_disc` is a pre-installed Python virtual environment with requirement Python packages.
+    - `-s configure` specified the building stage.
+    - `--bridge-gcc` and `--compiler-gcc` specified the GCC version building with tao bridge and disc compiler core.
+
+1. step2: build tao bridge library
+
+    ``` bash
+    python scripts/python/tao_build.py /opt/venv_disc -s build_tao_bridge
+    ```
+
+    the above command generates a dynamic library on path `tao/build/libtao_ops.so` if works well.
+
+1. step3: build disc compiler core
+
+    ``` bash
+    python scripts/python/tao_build.py /opt/venv_disc -s build_tao_compiler
+    ```
+
+    the above command generates an executable file on path
+    `tf_community/bazel-bin/tensorflow/compiler/decoupling/tao_compiler_main` if works well.
+
+1. [option]step4: run unit tests
+
+    please feel free to skip this step if you just want to build a Python wheel package.
+
+    ```bash
+    python scripts/python/tao_build.py /op/env_disc -s test_tao_compiler
+    ```
+
+1. step5: build Python wheel package
+
```

Comment:
Step 2, 3, and 5 could be merged together. At least it could be better to provide a way to build with only one command.

Reply:
I also prefer to build the package with one command using `setup.py` in the future.
But actually, users should execute 4 commands to build tensorflow_bladedisc at least, I consider keeping the separate steps in the current codebase because it's easier to locate which step run failed.